ibis/expr/types/groupby.py

```diff
@@ -79,6 +79,10 @@ def __init__(
                 ),
             )
         )
+
+        if not self.by:
+            raise ValueError("The grouping keys list is empty!")
```

Comment:
`ValueError` seems like it lacks useful type information. What about raising an `com.IbisInputError`, which while not super informative itself, it conveys that there was an error caused by some input from the user.

Reply:
Agree, `com.IbisInputError` seems like a better option.